### ISSUE TITLE
 Doesn't work on SLES11 Linux

### ISSUE THREAD

**Issue by rkedward** [sublimehq/sublime_merge#363]
**Version info**

 - OS: SLES11 Linux
 - Build: 1103

**Description**

Application won't start.  It reports the following error:

```
% sublime_merge
sublime_merge: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by sublime_merge)
sublime_merge: /lib64/libc.so.6: version `GLIBC_2.15' not found (required by sublime_merge)
```

```
% ldd --version
ldd (GNU libc) 2.11.3
Copyright (C) 2009 Free Software Foundation, Inc.
```
Can a build be compiled to work with an older glibc version?  sublime text 3 works in this environment.

**Steps to reproduce**

Run the command in SLES11.

**Expected behavior**

The application starts without an error.

**Comment by wbond** [comment]
You mention that Sublime Text works on your machine. What build of Sublime Text are you using?

**Comment by rkedward** [comment]
version 3.1.1, Build 3176

**Comment by jskinner** [comment]
Build 1116 should work on SLES11 (ldd reports it only requires GLIBC_2.7), however I don't expect this to always be the case going forward

**Comment by rkedward** [comment]
Thanks, looking forward to trying it out when build 1116 releases.

**Comment by rkedward** [comment]
I just downloaded and installed build 1116.  It didn't work outputting the following:

```
Unable to load libgdk-3.so
Unable to load gdk_atom_name from libgdk-3.0.so
Unable to load gdk_cairo_create from libgdk-3.0.so
Unable to load gdk_cairo_get_clip_rectangle from libgdk-3.0.so
Unable to load gdk_cursor_new_for_display from libgdk-3.0.so  
Unable to load gdk_cursor_unref from libgdk-3.0.so            
Unable to load gdk_display_get_default from libgdk-3.0.so     
Unable to load gdk_display_get_pointer from libgdk-3.0.so     
Unable to load gdk_error_trap_pop from libgdk-3.0.so          
Unable to load gdk_error_trap_push from libgdk-3.0.so         
Unable to load gdk_keymap_get_for_display from libgdk-3.0.so  
Unable to load gdk_keymap_translate_keyboard_state from libgdk-3.0.so
Unable to load gdk_keymap_get_modifier_state from libgdk-3.0.so      
Unable to load gdk_keyval_to_unicode from libgdk-3.0.so              
Unable to load gdk_pixbuf_new_from_file from libgdk-3.0.so           
Unable to load gdk_screen_get_default from libgdk-3.0.so             
Unable to load gdk_screen_get_display from libgdk-3.0.so             
Unable to load gdk_screen_get_height from libgdk-3.0.so              
Unable to load gdk_screen_get_root_window from libgdk-3.0.so         
Unable to load gdk_screen_get_width from libgdk-3.0.so               
Unable to load gdk_screen_get_n_monitors from libgdk-3.0.so          
Unable to load gdk_screen_get_monitor_at_point from libgdk-3.0.so    
Unable to load gdk_screen_get_monitor_geometry from libgdk-3.0.so    
Unable to load gdk_screen_get_resolution from libgdk-3.0.so          
Unable to load gdk_screen_get_rgba_visual from libgdk-3.0.so         
Unable to load gdk_unicode_to_keyval from libgdk-3.0.so              
Unable to load gdk_window_get_frame_extents from libgdk-3.0.so       
Unable to load gdk_window_get_origin from libgdk-3.0.so              
Unable to load gdk_window_get_state from libgdk-3.0.so               
Unable to load gdk_window_invalidate_rect from libgdk-3.0.so         
Unable to load gdk_window_set_cursor from libgdk-3.0.so              
Unable to load gdk_window_move_resize from libgdk-3.0.so             
Unable to load gdk_x11_display_get_type from libgdk-3.0.so           
Unable to load gdk_x11_display_get_xdisplay from libgdk-3.0.so       
Unable to load gdk_x11_window_get_xid from libgdk-3.0.so             
Unable to load gdk_x11_get_server_time from libgdk-3.0.so            
Unable to load gdk_x11_get_xatom_by_name_for_display from libgdk-3.0.so
Unable to load gdk_x11_window_set_user_time from libgdk-3.0.so         
Unable to load gdk_window_get_width from libgdk-3.0.so                 
Unable to load gdk_window_get_height from libgdk-3.0.so                
Unable to load gdk_drag_status from libgdk-3.0.so                      
Unable to load gdk_drag_context_get_actions from libgdk-3.0.so         
Unable to load gdk_drag_context_list_targets from libgdk-3.0.so        
Unable to load libgtk-3.so                                             
Unable to load gtk_accel_group_new from libgtk-3.so                    
Unable to load gtk_accelerator_get_default_mod_mask from libgtk-3.so   
Unable to load gtk_label_get_type from libgtk-3.so                     
Unable to load gtk_label_set_text from libgtk-3.so                     
Unable to load gtk_label_get_layout from libgtk-3.so                   
Unable to load gtk_label_get_layout_offsets from libgtk-3.so           
Unable to load gtk_bin_get_child from libgtk-3.so                      
Unable to load gtk_bin_get_type from libgtk-3.so                       
Unable to load gtk_box_get_type from libgtk-3.so                       
Unable to load gtk_box_pack_start from libgtk-3.so                     
Unable to load gtk_drawing_area_new from libgtk-3.so                   
Unable to load gtk_check_menu_item_get_type from libgtk-3.so           
Unable to load gtk_check_menu_item_new from libgtk-3.so                
Unable to load gtk_check_menu_item_new_with_label from libgtk-3.so     
Unable to load gtk_check_menu_item_set_active from libgtk-3.so         
Unable to load gtk_clipboard_clear from libgtk-3.so                    
Unable to load gtk_clipboard_get from libgtk-3.so                      
Unable to load gtk_clipboard_set_text from libgtk-3.so                 
Unable to load gtk_clipboard_set_with_data from libgtk-3.so            
Unable to load gtk_clipboard_store from libgtk-3.so                    
Unable to load gtk_clipboard_wait_for_text from libgtk-3.so            
Unable to load gtk_container_add from libgtk-3.so                      
Unable to load gtk_container_get_children from libgtk-3.so             
Unable to load gtk_container_get_type from libgtk-3.so                 
Unable to load gtk_container_remove from libgtk-3.so                   
Unable to load gtk_dialog_add_button from libgtk-3.so                  
Unable to load gtk_dialog_get_type from libgtk-3.so                    
Unable to load gtk_dialog_run from libgtk-3.so                         
Unable to load gtk_dialog_set_default_response from libgtk-3.so        
Unable to load gtk_drag_dest_set from libgtk-3.so                      
Unable to load gtk_drag_dest_add_text_targets from libgtk-3.so         
Unable to load gtk_drag_finish from libgtk-3.so                        
Unable to load gtk_file_chooser_add_filter from libgtk-3.so            
Unable to load gtk_file_chooser_dialog_new from libgtk-3.so            
Unable to load gtk_file_chooser_get_filename from libgtk-3.so          
Unable to load gtk_file_chooser_get_files from libgtk-3.so             
Unable to load gtk_file_chooser_get_type from libgtk-3.so              
Unable to load gtk_file_chooser_set_current_folder from libgtk-3.so    
Unable to load gtk_file_chooser_set_current_name from libgtk-3.so      
Unable to load gtk_file_chooser_set_do_overwrite_confirmation from libgtk-3.so
Unable to load gtk_file_chooser_set_local_only from libgtk-3.so               
Unable to load gtk_file_chooser_set_select_multiple from libgtk-3.so          
Unable to load gtk_file_filter_add_pattern from libgtk-3.so                   
Unable to load gtk_file_filter_new from libgtk-3.so                           
Unable to load gtk_file_filter_set_name from libgtk-3.so                      
Unable to load gtk_get_current_event_time from libgtk-3.so                    
Unable to load gtk_im_context_filter_keypress from libgtk-3.so                
Unable to load gtk_im_context_focus_in from libgtk-3.so                       
Unable to load gtk_im_context_focus_out from libgtk-3.so                      
Unable to load gtk_im_context_set_client_window from libgtk-3.so              
Unable to load gtk_im_context_set_cursor_location from libgtk-3.so            
Unable to load gtk_im_context_set_use_preedit from libgtk-3.so                
Unable to load gtk_im_multicontext_new from libgtk-3.so                       
Unable to load gtk_init from libgtk-3.so                                      
Unable to load gtk_init_check from libgtk-3.so                                
Unable to load gtk_main from libgtk-3.so                                      
Unable to load gtk_main_quit from libgtk-3.so                                 
Unable to load gtk_menu_attach_to_widget from libgtk-3.so                     
Unable to load gtk_menu_bar_new from libgtk-3.so                              
Unable to load gtk_menu_get_type from libgtk-3.so                             
Unable to load gtk_menu_item_get_label from libgtk-3.so                       
Unable to load gtk_menu_item_get_submenu from libgtk-3.so                     
Unable to load gtk_menu_item_get_type from libgtk-3.so                        
Unable to load gtk_menu_item_new from libgtk-3.so                             
Unable to load gtk_menu_item_new_with_label from libgtk-3.so                  
Unable to load gtk_menu_item_set_label from libgtk-3.so                       
Unable to load gtk_menu_item_set_submenu from libgtk-3.so                     
Unable to load gtk_menu_item_set_use_underline from libgtk-3.so               
Unable to load gtk_menu_new from libgtk-3.so                                  
Unable to load gtk_menu_popup from libgtk-3.so                                
Unable to load gtk_menu_shell_append from libgtk-3.so                         
Unable to load gtk_menu_shell_get_type from libgtk-3.so                       
Unable to load gtk_message_dialog_new from libgtk-3.so                        
Unable to load gtk_message_dialog_new_with_markup from libgtk-3.so            
Unable to load gtk_selection_data_get_uris from libgtk-3.so                   
Unable to load gtk_selection_data_get_format from libgtk-3.so                 
Unable to load gtk_selection_data_get_length from libgtk-3.so                 
Unable to load gtk_selection_data_set_text from libgtk-3.so                   
Unable to load gtk_separator_menu_item_new from libgtk-3.so                   
Unable to load gtk_settings_get_default from libgtk-3.so                      
Unable to load gtk_show_uri from libgtk-3.so                                  
Unable to load gtk_vbox_new from libgtk-3.so                                  
Unable to load gtk_widget_add_accelerator from libgtk-3.so                    
Unable to load gtk_widget_add_events from libgtk-3.so                         
Unable to load gtk_widget_destroy from libgtk-3.so                            
Unable to load gtk_widget_get_display from libgtk-3.so                        
Unable to load gtk_widget_get_parent from libgtk-3.so                         
Unable to load gtk_widget_get_screen from libgtk-3.so                         
Unable to load gtk_widget_get_type from libgtk-3.so                           
Unable to load gtk_widget_get_window from libgtk-3.so                         
Unable to load gtk_widget_grab_focus from libgtk-3.so                         
Unable to load gtk_widget_hide from libgtk-3.so                               
Unable to load gtk_widget_remove_accelerator from libgtk-3.so                 
Unable to load gtk_widget_set_app_paintable from libgtk-3.so                  
Unable to load gtk_widget_set_double_buffered from libgtk-3.so                
Unable to load gtk_widget_set_sensitive from libgtk-3.so                      
Unable to load gtk_widget_set_visual from libgtk-3.so
Unable to load gtk_widget_show from libgtk-3.so
Unable to load gtk_widget_show_all from libgtk-3.so
Unable to load gtk_widget_size_request from libgtk-3.so
Unable to load gtk_widget_get_preferred_height from libgtk-3.so
Unable to load gtk_widget_get_allocation from libgtk-3.so
Unable to load gtk_widget_get_allocated_width from libgtk-3.so
Unable to load gtk_widget_get_allocated_height from libgtk-3.so
Unable to load gtk_widget_get_scale_factor from libgtk-3.so
Unable to load gtk_widget_get_style_context from libgtk-3.so
Unable to load gtk_widget_set_halign from libgtk-3.so
Unable to load gtk_widget_set_valign from libgtk-3.so
Unable to load gtk_widget_get_direction from libgtk-3.so
Unable to load gtk_widget_create_pango_layout from libgtk-3.so
Unable to load gtk_widget_path_append_type from libgtk-3.so
Unable to load gtk_widget_path_copy from libgtk-3.so
Unable to load gtk_widget_path_free from libgtk-3.so
Unable to load gtk_window_add_accel_group from libgtk-3.so
Unable to load gtk_window_fullscreen from libgtk-3.so
Unable to load gtk_window_get_type from libgtk-3.so
Unable to load gtk_window_iconify from libgtk-3.so
Unable to load gtk_window_maximize from libgtk-3.so
Unable to load gtk_window_move from libgtk-3.so
Unable to load gtk_window_new from libgtk-3.so
Unable to load gtk_window_present from libgtk-3.so
Unable to load gtk_window_present_with_time from libgtk-3.so
Unable to load gtk_window_remove_accel_group from libgtk-3.so
Unable to load gtk_window_resize from libgtk-3.so
Unable to load gtk_window_get_size from libgtk-3.so
Unable to load gtk_window_get_position from libgtk-3.so
Unable to load gtk_window_set_default_icon_list from libgtk-3.so
Unable to load gtk_window_set_default_size from libgtk-3.so
Unable to load gtk_window_set_keep_above from libgtk-3.so
Unable to load gtk_window_set_modal from libgtk-3.so
Unable to load gtk_window_set_position from libgtk-3.so
Unable to load gtk_window_set_title from libgtk-3.so
Unable to load gtk_window_set_transient_for from libgtk-3.so
Unable to load gtk_window_set_type_hint from libgtk-3.so
Unable to load gtk_window_stick from libgtk-3.so
Unable to load gtk_window_unfullscreen from libgtk-3.so
Unable to load gtk_widget_modify_bg from libgtk-3.so
Unable to load gtk_window_set_decorated from libgtk-3.so
Unable to load gtk_window_set_skip_taskbar_hint from libgtk-3.so
Unable to load gtk_window_set_skip_pager_hint from libgtk-3.so
Unable to load gtk_drag_begin from libgtk-3.so
Unable to load gtk_target_list_new from libgtk-3.so
Unable to load gtk_selection_data_get_text from libgtk-3.so
Unable to load gtk_style_context_new from libgtk-3.so
Unable to load gtk_style_context_save from libgtk-3.so
Unable to load gtk_style_context_restore from libgtk-3.so
Unable to load gtk_style_context_get_path from libgtk-3.so
Unable to load gtk_style_context_set_path from libgtk-3.so
Unable to load gtk_style_context_set_parent from libgtk-3.so
Unable to load gtk_render_layout from libgtk-3.so
```

**Comment by wbond** [comment]
I can't tell for sure, but it would seem to be that your distro is rather old (originally released 10 years ago), and doesn't have official GTK3 packages. In fact, I believe it was released before Sublime Text even ran on Linux (ST2). :-)

It even appears SLES11 is no longer supported by SUSE as of a couple of months ago. Unfortunately it seems you'll have to move to SLES12 to get GTK3 support. I found a post suggesting version 3.10 is available for SLES12.
